comment-coverage/src/index.js

```diff
@@ -33,9 +33,30 @@ function getSummaryType(fileName) {
     core.setFailed(`"The file ${fileName}" could not be linked to a known coverage report. Valid file name end in: "summary.json" (jest report), "cobertura.xml" (tarpaulin report) or "-coverage" (last line of go tool report)`)
 }
 
+async function getJsonContent(octokit, owner, repo, summary, label) {
+    try {
+        return await octokit.rest.repos.getContent({
+            owner,
+            repo,
+            path: `${summary.key}.json`,
+            ref: 'coverage',
+        }).then((result) => {
+            try {
+                const content = Buffer.from(result.data.content, 'base64').toString()
+                return JSON.parse(content)
+            } catch (e) {
+                return {label, coverage: 'nd', color: 'red'}
+            }
+        });
+    } catch (e) {
+        return {label, coverage: 'nd', color: 'red'}
+    }
```

Comment:
```suggestion
    try {
        const contentResult = await octokit.rest.repos.getContent({
            owner,
            repo,
            path: `${summary.key}.json`,
            ref: 'coverage',
        })
        const content = Buffer.from(contentResult.data.content, 'base64').toString()
        return JSON.parse(content)
    } catch (e) {
        return {label, coverage: 'nd', color: 'red'}
    }
```

I think we can simplify with this :)
I did not tried it because I did not want to commit on your branch

Reply:
Way better, thanks!